Userland/Libraries/LibAudio/Resampler.h

```diff
@@ -62,7 +62,8 @@ class ResampleHelper {
     template<ArrayLike<SampleType> Samples, size_t vector_inline_capacity = 0>
     ErrorOr<void> try_resample_into_end(Vector<SampleType, vector_inline_capacity>& destination, Samples&& to_resample)
     {
-        TRY(destination.try_ensure_capacity(destination.size() + to_resample.size() * ceil_div(m_source, m_target)));
+        float ratio = (m_source > m_target) ? static_cast<float>(m_source) / m_target : static_cast<float>(m_target) / m_source;
+        TRY(destination.try_ensure_capacity(destination.size() + to_resample.size() * ratio));
```

Comment:
Hmmm, was this only fixed in one of the other resampler functions?

Reply:
Not sure what you mean here. I'm not sure when the old behavior would ever be correct, since it's doing integer math. playing the 22050 song in aplay plays it with huge gaps in the playback, while playing in sound player is glitchy because the "Resample" thing fails to actually fill the destination with 2x the samples (the logic expects m_source > m_target always).

Reply:
I meant that I'm pretty sure I fixed this kind of issue already at some point, so it must have been in another resampling function. I should actually look up how to do proper band-limited (or band-extended) FIR resampling and implement that...